docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
+
+### Ui
+The `Ui` class is responsible for the user interface of the application. It receives the input from the user and prints the results to the user.
+
+### Parser
+
+The `Parser` class is responsible for interpreting the user's input and calling the respective command. 
+
+### Command
+
+The different `Command` classes are responsible for receiving the user's input from the `Parser` class and executing the corresponding commands. 
+The figure belows shows the class diagram of the command class: <br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/command.JPG" alt="" width="200"/><br/>
+All Command classes inherit from the `Command` class.
+
+### Storage
+
+The `Storage` class is responsible for reading and writing data to the hard disk.
+
+
+### Model
+The model is responsible for holding the data of the application in memory. The figure below shows the structure of the Model in this application.
+
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/Model_Class_Diagram.JPG" alt="" width="750"/>
+
+ When executing commands,the `CalendarItem` class updates the information or provides the information of different types of items if needed. 
+It is split into two subclasses: `Task` and `Event` class.
+The `CalendarList` class holds the array of `CalendarItem` and keeps track of the number of total items, total tasks, and total events.
+
+The `Task` and `Event` classes are further explained below.
+
+#### Task
+
+The `Task` class stores the information of all task items, including description and status, such as isDone and isImportant. 
+It has two subclasses: `Todo` and `Deadline`. `Deadline` items also have the date information and countdown for the deadline date, which are not included in `Todo` tasks. 
+
+#### Event
+
+The `Event` class stores the information of all `Event` items, including date, time, venue, status whether it is overdue, and any other information if added. 
+It has two subclasses: SchoolEvent and Activity. `Activity` items can have other details.
+All `SchoolEvent` items must have a module code.  
+Furthermore, the `SchoolEvent` class has four subclasses which are `Lecture`, `Tutorial`, `Lab`, and `Exam`. Among them only `Exam` items have a countdown for the exam date. 
+All `Event` items update the information or provide needed information about `Event` items when executing commands related to events or saving the information to `Storage`.
+
+Design consideration: `SchoolEvent` is modelled after NUS modules to cater to our intended users.
+
+## Implementation
+This section describes the details of how each feature is implemented.
+
+### Add a calendar item feature
+This feature is facilitated by `AddCommand`.
+The following sequence diagram shows how the `execute()` operation works:<br/>
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/add_command_SD.JPG" alt="" width="750"/><br/>
```

Comment:
![Screenshot 2020-10-30 124101](https://user-images.githubusercontent.com/13282967/97660665-383b3200-1aad-11eb-8ac0-2c24dea46db6.png)
I think the line used to separate the alt frame has to be a dotted line instead of a solid line.